mentat/parsers/streaming_printer.py

```diff
@@ -0,0 +1,78 @@
+import asyncio
+from collections import deque
+from typing import Any, Dict, List, Tuple
+
+from mentat.session_context import SESSION_CONTEXT
+
+FormattedString = str | Tuple[str, Dict[str, Any]] | List[Tuple[str, Dict[str, Any]]]
```

Comment:
Might be better to make `FormattedString` an actual class instead of a union of all these types.

Reply:
Maybe; I'll add a TODO but I want to get this PR merged quickly because I need it in order to work on my textual PR.